erts/emulator/nifs/common/prim_file_nif.c

```diff
@@ -414,6 +419,66 @@ static void owner_death_callback(ErlNifEnv* env, void* obj, ErlNifPid* pid, ErlN
     }
 }
 
+static void dyncall_dup(ErlNifEnv* env, efile_data_t* d, struct prim_file_nif_dyncall_dup *dc_dup) {
+
+    enum efile_state_t previous_state;
+
+    previous_state = erts_atomic32_cmpxchg_acqb(&d->state,
+        EFILE_STATE_BUSY, EFILE_STATE_IDLE);
+
+    if (previous_state == EFILE_STATE_IDLE) {
+        int do_dup = !0;
+
+        dc_dup->error = efile_get_handle(env, d, do_dup, &dc_dup->handle);
+
+        previous_state = erts_atomic32_cmpxchg_relb(&d->state,
+            EFILE_STATE_IDLE, EFILE_STATE_BUSY);
+
+        ASSERT(previous_state != EFILE_STATE_IDLE);
+
+        if(previous_state == EFILE_STATE_CLOSE_PENDING) {
+            /* This is the only point where a change from CLOSE_PENDING is
```

Comment:
Nitpick: this is done in `file_handle_wrapper` too, so both comments need to be updated (there's nothing wrong with the code though).

Reply:
Oh.  So now neither place is the only point... Right!

Reply:
It would be nice to merge those two functions, to remove code duplication, but I did not see an elegant way.